src/server.cpp

```diff
@@ -240,6 +240,22 @@ Server::Server(
 
 	if (!gamespec.isValid())
 		throw ServerError("Supplied invalid gamespec");
+
+#if USE_PROMETHEUS
+	m_metrics_backend = std::unique_ptr<MetricsBackend>(new PrometheusMetricsBackend());
+#else
+	m_metrics_backend = std::unique_ptr<MetricsBackend>(new MetricsBackend());
+#endif
+
+	m_uptime_gauge = m_metrics_backend->addGauge("core_server_uptime", "Server uptime (in seconds)");
+	m_player_gauge = m_metrics_backend->addGauge("core_player_number", "Number of connected players");
+	m_step_counter = m_metrics_backend->addCounter("core_server_step_time", "Server step time (in milliseconds)");
```

Comment:
how do I use this as a useful value without knowing the number of steps?

Reply:
you derivate the step counter and you have the mean step time over a period

Reply:
And how do you do that without guessing based on the step setting?